test/unit/spec/common/jobStateViewer-spec.js

```diff
@@ -0,0 +1,349 @@
+define([
```

Comment:
Tests for both modules came from taking apart the existing job log viewer tests and reusing the relevant tests, and I then added new tests for additional features/functionality.